src/test/java/htsjdk/samtools/CRAMFileWriterTest.java

```diff
@@ -132,40 +141,48 @@ private void writeRecordsToCRAM(CRAMFileWriter writer, List<SAMRecord> samRecord
-        CRAMFileReader cReader = new CRAMFileReader(null, is, referenceSource);
-
-        SAMRecordIterator iterator2 = cReader.getIterator();
-        int index = 0;
-        while (iterator2.hasNext()) {
-            SAMRecord actualRecord = iterator2.next();
-            SAMRecord expectedRecord = expectedRecords.get(index++);
-
-            Assert.assertEquals(actualRecord.getReadName(), expectedRecord.getReadName());
-            Assert.assertEquals(actualRecord.getFlags(), expectedRecord.getFlags());
-            Assert.assertEquals(actualRecord.getAlignmentStart(), expectedRecord.getAlignmentStart());
-            Assert.assertEquals(actualRecord.getAlignmentEnd(), expectedRecord.getAlignmentEnd());
-            Assert.assertEquals(actualRecord.getReferenceName(), expectedRecord.getReferenceName());
-            Assert.assertEquals(actualRecord.getMateAlignmentStart(),
-                    expectedRecord.getMateAlignmentStart());
-            Assert.assertEquals(actualRecord.getMateReferenceName(),
-                    expectedRecord.getMateReferenceName());
-            Assert.assertEquals(actualRecord.getReadBases(), expectedRecord.getReadBases());
-            Assert.assertEquals(actualRecord.getBaseQualities(), expectedRecord.getBaseQualities());
+        try (CRAMFileReader cReader = new CRAMFileReader(null, is, referenceSource)) {
+
+            SAMRecordIterator iterator2 = cReader.getIterator();
+            int index = 0;
+            while (iterator2.hasNext()) {
+                SAMRecord actualRecord = iterator2.next();
+                SAMRecord expectedRecord = expectedRecords.get(index++);
+
+                Assert.assertEquals(actualRecord.getReadName(), expectedRecord.getReadName());
+                Assert.assertEquals(actualRecord.getFlags(), expectedRecord.getFlags());
+                Assert.assertEquals(actualRecord.getAlignmentStart(), expectedRecord.getAlignmentStart());
+                Assert.assertEquals(actualRecord.getAlignmentEnd(), expectedRecord.getAlignmentEnd());
+                Assert.assertEquals(actualRecord.getReferenceName(), expectedRecord.getReferenceName());
+                Assert.assertEquals(actualRecord.getMateAlignmentStart(),
+                        expectedRecord.getMateAlignmentStart());
+                Assert.assertEquals(actualRecord.getMateReferenceName(),
+                        expectedRecord.getMateReferenceName());
+                Assert.assertEquals(actualRecord.getReadBases(), expectedRecord.getReadBases());
+                Assert.assertEquals(actualRecord.getBaseQualities(), expectedRecord.getBaseQualities());
+
+                Assert.assertEquals(actualRecord.getAttributes().stream().map(s -> s.tag).collect(Collectors.toSet()),
+                        actualRecord.getAttributes().stream().map(s -> s.tag).collect(Collectors.toSet()), expectedRecord.getReadName());
```

Comment:
Isn't this comparing actual to actual?  I think there was a typo.

Reply:
oops